controllers/index.js

```diff
@@ -2,35 +2,267 @@
 
 
 var Product = require('../models/productModel');
+var async = require('async');
+var paypal = require('paypal-rest-sdk');
 
 
-module.exports = function (router) {
+module.exports.index = function (req, res, next) {
+	modelBuilder(req, res, function (err, model) {
+		if (err) {
+			return next(err);
+		}
+		res.render(req.url, model);
+	});
+
+};
+
+var modelBuilder = function (req, res, callback) {
+	var getCart = function(callback) {
+		var displayCart = {items: [], total: 0};
+		var sessionCart = req.session.cart;
+		var total = 0;
+		var totalItems = 0;
+		var model = {
+			items: null,
+			result: null,
+			continueMessage: null,
+			totalItems:0
+
+		};
+		var cartLength = 0;
+		if (!sessionCart) {
+			model.result = res.bundle.get('empty');
+			model.continueMessage = res.bundle.get('keepShopping');
+
+		}
+		else {
+			//Ready the products for display
+			Object.keys(sessionCart).forEach(function (itemid) {
+				var item = sessionCart[itemid];
+				cartLength += item.qty;
+				displayCart.items.push(item);
+				total += (item.qty * item.price);
+				totalItems += item.qty;
+			});
 
+			req.session.total = displayCart.total = total.toFixed(2);
+			model.items = displayCart;
+			model.totalItems = totalItems;
+			model.itemsInCart = res.bundle.getIntl('items', {cartItemLength: cartLength});
+		}
 
-	router.get('/', function (req, res) {
+		callback(null, model);
+	};
 
+	var getProducts = function (callback) {
 		Product.find(function (err, prods) {
 			if (err) {
 				console.log(err);
 			}
-			prods.forEach(function (prod) {
-				prod.prettyPrice = prod.prettyPrice();
+			prods = prods.map(function (prod) {
+				prod.prettyPrice = prod.prettifyPrice();
+				return prod;
 			});
+			var staticProd = [{
+				prettyPrice: '$100.00',
+				name: 'foo',
+				id: 1
+			}];
 			var model = {
 				products: prods,
 				title: 'Kraken Store'
 			};
-			res.render('index', model);
+			//console.log('model', model);
+			callback(null, model);
 		});
+	};
+	async.parallel({cart: getCart, products: getProducts}, function (err, result) {
```

Comment:
assuming either was a longish running process (they aren't in this isolated case) it could be a good pattern to use for illustrative purposes. that was my thought.